### PR TITLE
Adding Model.removeRelationship() and ContainerInstance.removeEfferentRelationshipWith()

### PR DESCRIPTION
**Pull request by mjiderhamn**
I found myself needing to correct Structurizrs assumption that all container instances have a relationship to all instances of a container they are using. Could this be integrated into the core version...?

### CODE REVIEW

File: structurizr-core/src/com/structurizr/model/Model.java [hunk 1]
```diff
@@ -199,6 +199,12 @@ private void addRelationshipToInternalStructures(Relationship relationship) {
         idGenerator.found(relationship.getId());
     }
 
+    /** Remove existing {@link Relationship} */
+    public void removeRelationship(Relationship relationship) {
```
Comment (mjiderhamn) [inline structurizr-core/src/com/structurizr/model/Model.java]: Maybe this should be package protected instead?

**Comment by simonbrowndotje** [comment]
What's the problem you're trying to solve? Are you getting too many relationships on a diagram?

**Comment by mjiderhamn** [comment]
In my specific case I (unfortunately) have services `I` and `R` on two deployment nodes and service `T` on a third node. `I` uses `R` *on the same node* as itself and `T` uses a specific instance of `R`.

Out of the box, Structurizr assumes all `R`s are used by all `I`s and the `T`.

**Comment by mjiderhamn** [comment]
Using this PR
```
i1.removeEfferentRelationshipWith(i2);
i2.removeEfferentRelationshipWith(i1);
t.removeEfferentRelationshipWith(r2);
```
solves the issue.

**Comment by simonbrowndotje** [comment]
Right, understood, I can see why you'd want to remove those relationships. I'd really like to try and keep the model as append only (e.g. removing relationships after views have been created will cause issues) ... perhaps adding another method, e.g. DeploymentNode.add(Container, true|false), would be a better idea, with that last flag indicating whether the container-container relationships should automatically be replicated between container instances. What do you think?

**Comment by mjiderhamn** [comment]
I think that makes sense. In my case it'll probably add up to about equal amount of code.

I'm looking at two other issues, which I'd be thankful if you could give me a hint would be related to this new approach, or if it something completely separate. I want to be able to determine the "magnification level", so to speak.
* In one case I want to create a deployment view, but only include deployment nodes N levels down, or only nodes matching (or not matching) some filter criteria. For example I'd like to create a view of all the (virtual) machines included in a system, without caring about the software running on each (or including top level software such as application servers, but not their containers/applications).
* In another case I want to include a deployment node in a deployment view, without including its parent and definitely not the siblings. The case is that I want to create a partial deployment view, where containers on machine A have a relationship to 1 out of many containers running on machine B.  The whole point is to focus on machine A without being confused by everything on machine B, but as it is now, I can only include everything on B exclude it altogether. (Which has left me with the workaround of creating a "dummy B" with only the relevant service, which kind of defeats the purpose of the consistent model.)

**Comment by simonbrowndotje** [comment]
Thanks, I'll close this PR then and add an alternative version of the add(Container) method.

The library, and specifically the DeploymentView, isn't really designed to show what you want to show, so perhaps you're best off creating a custom PlantUML renderer to filter out the deployment details you don't want to show.

**Pull request closed without merging**